[ticket/13867] Manage new profile field types

mrgoldy opened the pull request:
Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-13867

---

Continuation of #4603 ❤️

rxu commented:
Currently custom profile fields system is missing handling of CPF icons for post miniprofiles. For the built-in "custom" fields icons are just hardcoded, for all user defined CPFs simply the same default icon is used from `/styles/[stylename]/theme/images/icons_contact.png`.
So it would be nice to add the feature with this PR if possible as it touches CPFs alltogether :)

mrgoldy commented:
While I agree that is something that might need to be handled, but that's not what this PR is aimed to do. This PR was originally created by @javiexin in 2015 and sat on the shelf ever since.

phpBB has custom profile field types, such as 'boolean', 'string', 'date', etc..
These are extendible by extensions. However, when an extension adds a new type and that extension is later on disabled and deleted, all profile fields using that type will throw errors all over the board. The methods introduced in this PR allow handling those situation. Much like the extension manager allows handling of the notification types inside the `ext.php` file.
This PR provides pretty much the same functionality for profile fields.

It does indeed touch the CPF manager, mostly cause I cleaned up the code. 
I did not alter any existing functionality however, everything is still exactly the same.
I wouldn't mind adding to this PR, but seeing this has been postponed and postponed some more, I would argue that it might be better to first 'just get this one in', and then continue on it by creating a separate PR's for handling any CPF icons.

dark-1 commented:
With regards to this ticket, I am with @mrgoldy ,
coz this ticket has been for a very long time & needs to be fulfilled as-soon-as-possible.

Adding new feature can be done via new Ticket/PR...

Best regards 👍

dark-1 commented:
Somebody needs to restart this job : https://ci.appveyor.com/project/phpBB/phpbb/builds/31667572
Err0r : 
```
GitHub API limit (0 calls/hr) is exhausted, could not fetch https://api.github.com/repos/bantuXorg/php-ini-get-wrapper/zipball/4770c7feab370c62e23db4f31c112b7c6d90aee2. Create a GitHub OAuth token to go over the API rate limit. You can also wait until ? for the rate limit to reset.
Head to https://github.com/settings/tokens/new?scopes=repo&description=Composer+on+APPVYR-WIN+2020-03-24+0349
to retrieve a token. It will be stored in "C:/Users/appveyor/AppData/Roaming/Composer/auth.json" for future use by Composer.
Token (hidden): 
```

Best regards 👍

rxu commented on tests/profilefields/manager_test.php:
```diff
@@ -0,0 +1,178 @@
+		return $this->createXMLDataSet(dirname(__FILE__).'/fixtures/manager.xml');
+	}
+
+	public function setUp()
```
Looks like it's broke tests on 3.3/master, should be `public function setUp(): void` probably.
